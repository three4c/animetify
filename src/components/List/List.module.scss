.List {
  & &__header {
    margin-bottom: 48px;
  }

  & &__title {
    display: inline-block;
    margin-bottom: 16px;
  }

  & &__works {
    display: block;
    font-weight: 400;
  }

  & &__masonry {
    display: flex;
  }

  & &__column {
    background-clip: padding-box;
    width: 272px !important;

    &:not(:last-child) {
      margin-right: 24px;
    }
  }

  & &__noSong {
    font-size: 1.4rem;
  }
}
