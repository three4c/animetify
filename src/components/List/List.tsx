import React from 'react';
import Masonry from 'react-masonry-css';
import { AnisonProgramList } from '../../types';

import Card from '../Card/Card';
import Song from '../Song/Song';

import styles from './List.module.scss';

export interface ListProps {
  /** アニソン API */
  works: {
    animeName: string;
    animeSong: AnisonProgramList[];
    broadcastStartDate: string;
  }[];
  /** 年 */
  year?: string;
  /** 季節 */
  season?: '春' | '夏' | '秋' | '冬';
  /** タイトル */
  title?: string;
}

const List: React.FC<ListProps> = (props) => {
  const breakpointColumnsObj = {
    default: 5,
    1456: 4,
    1160: 3,
  };

  return (
    <div className={styles.List}>
      <div className={styles.List__header}>
        <h1 className={styles.List__title}>
          {props.works.length === 0
            ? '見つかりませんでした :('
            : props.year && props.season
            ? `${props.year}年の${props.season}アニメ`
            : props.title}
        </h1>
        {props.works.length !== 0 && (
          <span
            className={styles.List__works}
          >{`${props.works.length}作品`}</span>
        )}
      </div>
      {props.works.length !== 0 && (
        <Masonry
          breakpointCols={breakpointColumnsObj}
          className={styles.List__masonry}
          columnClassName={styles.List__column}
        >
          {props.works.map((item, index) => {
            return (
              <Card
                key={index}
                title={item.animeName}
                broadcastStartDate={item.broadcastStartDate}
              >
                <Song items={item.animeSong} />
              </Card>
            );
          })}
        </Masonry>
      )}
    </div>
  );
};

export default List;
