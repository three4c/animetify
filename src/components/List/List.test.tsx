import React from 'react';

import { shallow } from 'enzyme';

import List from './List';

describe('<List />', () => {
  const works = [
    {
      animeName: '電波女と青春男',
      broadcastStartDate: '2011-04-15',
      animeSong: [
        {
          animeName: '電波女と青春男',
          animeClassification: 'TV',
          musicName: 'Os-宇宙人',
          artistName: '大亀あすか',
          animeDescription: 'OP',
          broadcastOrder: '',
          broadcastStartDate: '2011-04-15',
        },
        {
          animeName: '電波女と青春男',
          animeClassification: 'TV',
          musicName: 'ルル',
          artistName: 'やくしまるえつこ',
          animeDescription: 'ED',
          broadcastOrder: '',
          broadcastStartDate: '2011-04-15',
        },
      ],
    },
  ];

  it('SnapShot（works）', () => {
    expect(shallow(<List works={works} />)).toMatchSnapshot();
  });

  it('SnapShot（year, season）', () => {
    expect(
      shallow(<List works={works} year="2011" season="春" />)
    ).toMatchSnapshot();
  });

  it('SnapShot（Not Found）', () => {
    expect(shallow(<List works={[]} />)).toMatchSnapshot();
  });
});
