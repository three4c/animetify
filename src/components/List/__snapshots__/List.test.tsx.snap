// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> SnapShot（Not Found） 1`] = `
<div
  className="List"
>
  <div
    className="List__header"
  >
    <h1
      className="List__title"
    >
      見つかりませんでした :(
    </h1>
  </div>
</div>
`;

exports[`<List /> SnapShot（works） 1`] = `
<div
  className="List"
>
  <div
    className="List__header"
  >
    <h1
      className="List__title"
    />
    <span
      className="List__works"
    >
      1作品
    </span>
  </div>
  <Masonry
    breakpointCols={
      Object {
        "1160": 3,
        "1456": 4,
        "default": 5,
      }
    }
    className="List__masonry"
    columnClassName="List__column"
  >
    <Card
      broadcastStartDate="2011-04-15"
      key="0"
      title="電波女と青春男"
    >
      <Memo(Song)
        items={
          Array [
            Object {
              "animeClassification": "TV",
              "animeDescription": "OP",
              "animeName": "電波女と青春男",
              "artistName": "大亀あすか",
              "broadcastOrder": "",
              "broadcastStartDate": "2011-04-15",
              "musicName": "Os-宇宙人",
            },
            Object {
              "animeClassification": "TV",
              "animeDescription": "ED",
              "animeName": "電波女と青春男",
              "artistName": "やくしまるえつこ",
              "broadcastOrder": "",
              "broadcastStartDate": "2011-04-15",
              "musicName": "ルル",
            },
          ]
        }
      />
    </Card>
  </Masonry>
</div>
`;

exports[`<List /> SnapShot（year, season） 1`] = `
<div
  className="List"
>
  <div
    className="List__header"
  >
    <h1
      className="List__title"
    >
      2011年の春アニメ
    </h1>
    <span
      className="List__works"
    >
      1作品
    </span>
  </div>
  <Masonry
    breakpointCols={
      Object {
        "1160": 3,
        "1456": 4,
        "default": 5,
      }
    }
    className="List__masonry"
    columnClassName="List__column"
  >
    <Card
      broadcastStartDate="2011-04-15"
      key="0"
      title="電波女と青春男"
    >
      <Memo(Song)
        items={
          Array [
            Object {
              "animeClassification": "TV",
              "animeDescription": "OP",
              "animeName": "電波女と青春男",
              "artistName": "大亀あすか",
              "broadcastOrder": "",
              "broadcastStartDate": "2011-04-15",
              "musicName": "Os-宇宙人",
            },
            Object {
              "animeClassification": "TV",
              "animeDescription": "ED",
              "animeName": "電波女と青春男",
              "artistName": "やくしまるえつこ",
              "broadcastOrder": "",
              "broadcastStartDate": "2011-04-15",
              "musicName": "ルル",
            },
          ]
        }
      />
    </Card>
  </Masonry>
</div>
`;
